lib/orderbook/OrderBook.ts

```diff
@@ -201,9 +204,20 @@ class OrderBook extends EventEmitter {
     const pair = this.pairs.get(pairId);
     if (pair) {
       this.pairs.delete(pairId);
-      this.matchingEngines.delete(pairId);
-      // TODO: invalidate all orders for this pair
-      // TODO: update handshake state
+      const matchingEngine = this.matchingEngines.get(pairId);
+
+      if (matchingEngine) {
+        if (this.pool) {
+          matchingEngine.ownOrders.buy.forEach(this.pool.broadcastOrderInvalidation);
+          matchingEngine.ownOrders.sell.forEach(this.pool.broadcastOrderInvalidation);
```

Comment:
didn't we say we want to allow to invalidate a pair, instead of broadcasting a packet for each order separately?

Reply:
Good idea!